Rust Heretic rework: THE PUKENING.

EnterTheJake opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments; otherwise, it may not be viewable. -->
<!You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Turf rusting is now based on knowledge instead of RNG; toxic damage on spells has been replaced with disgust; rust walkers are a bit easier to summon and show up early in the tree.

Rust ascension spread has been massively increased  and immunities while on Rust have been updated.
## Why It's Good For The Game

This is a massive list of changes, but the tl:dr is Rust heretic design is pretty barebones and dated in its current state. I'm going to try my best to explain why that is and why it needs to change.

###  RNG on rusting is bad

**Problem** : quite a lot of turfs cannot be rusted, and reinforced turfs rusting is tied to RNG, the emergency shuttle floors are also unrustable, making an ascended heretic not threatening at all to an escaping crew.

**Solution**: Heretic rusting prowess is now tied to knowledge progression; reinf wall rusting is no longer tied to RNG; the emergency shuttle is now rustable if you have accrued enough knowledge.

**Explanation** : I've been adopting the code from a previous PR that sadly didn't make the light of day.

Link: https://github.com/tgstation/tgstation/pull/65361

The goal is to have Heretic rusting be more consistent, so that if you ascend the shuttle is no longer a safe refuge.

**Grasp Of Rust**: Basic floors and walls.
**Mark Of Rust**: Reinf floors and walls (takes 3 applications of rust to fully destroy).
**Toxic Blade** Titanium and Plastitanium( still takes 3 applications of rust to destroy).
**Ascension**: almost everything save for admin walls, glass, and silver.

### A territorial antagonist with no tools to defend its turf

**Problem**: Rust heretic is forced into a territorial playstyle, but it's heavily diminished by rusted turfs not having any negative effect on the crew whatsoever, Toxic damage on plume and blade is also kind of bad and has no synergy with the rest of the kit.

Lastly, while fitting to the theme of Rust; mark detonation destroying clothes and items can end up being more harmful to the heretic, as destroying a secoff's flashbang will result in you getting stunned and killed.

**Solution**: Rusted floors are no longer safe for the crew; walking over Rust will add a small buildup of disgust, silicons will receive ticking damage, Toxic damage on Rust skills has been replaced with disgust buildup and a chem purge effect to prevent the crew from cheesing the debuff  with sol dry.

**Explaination** "The goal behind this change is to prevent the crew to just be able to walk willy nilly into an eldritchly decayed area without repercussions, plus I feel like disgust is a fitting debuff for Rusties for a number of reasons.

1) Lore wise, your goal is to turn the entire station into a garbage dump.

2) It fits for fighting a territorial antagonist, it's not immediately threatening on its own, but if you overstay your welcome, it has dire consequences. 

3) We don't really do much with disgust as a status effect, far as i know, this'll be the first iteration of it being used for an antagonist toolset.

4) It gives reason to the crew to act hostile towards the Heretic, as you are essentially making their work area inhabitable.

I've also Slightly increased the healing on leeching walk and gave it some minor temperature regulation (you won't be able to outtemp space cold).

### Rust Walkers 

**Problem**: Rust walkers might as well not exist in their current state; they are one of the latest heretic unlocks, and the recipe for summoning one is INSANE now that you cannot decapitate mobs anymore.

**Solution**: Rust walkers pop up a bit earlier in the tree; the ritual to summon them has more reasonable reagents (wires, pools of vomit, iron sheets) and they have increased health.

**Explanation**: : There is not a lot to add to this. without walkers, Rust heretics are fairly slow at spreading rust, by making rust walkers have reasonable summoning reagents, we ease that problem; lastly i've slightly bumped their hp, as i felt 75 hp was a bit low.

### Underwhelming ascension

**Problem**: The spread rate from the ascension is downright ATROCIOUS in its current state, i actually ran multiple tests on different maps, on Metastation it can take up to 40 minutes for the rust to spread to the entire station, considering an ascended heretic results in an immediate shuttle call, it's unecceptable in its current form.

Solution: Rust spread is now a hybridization between old ascension and Wizard tranformation ritual.

**Explanation**: The spread rate in its current form is painstakingly slow and has a tendency to reach one corner of the map and compound on itself, i've borrowed the code from the wizard final ritual "transformation" and gave it my own spin  so that given enough time it will cover the station more or less evenly, it can now spread through all z levels.

Video example: https://www.youtube.com/watch?v=jZ5zMrNM6Jw

I've also updated the immunities you gain on ascension to ignore slowdown and stasis from Cryogelidia.

The whole point of rust ascension is to become immune to crowd control, i've seen far too many ascended heretics die to a single bola or cryo syringe, that's pretty lame so i fixed that.

Lastly, aggressive spread has had its radius reduced in exchange for the spread being consistent now; cooldown is also halved when you ascend, to better help the heretic go on the offensive.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far-reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or administrators, you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the TGstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Rust Heretics rusting is now consistent and tied to knowledge progression.
balance: Walking on rust applies disgust and chem purge to non-heretics and brute damage to silicons.
balance: Toxic damage on heretic skills is replaced with disgust.
balance: Aggressive Spread radius is now a bit shorter in exchange for losing its RNG elements, cooldown is halved on ascension.
balance: Pulse of Entropy had its radius doubled and the recipe simplified.
balance: Leeching walk has had its healing increased and a minor temp regulation effect added.
balance: Rust Walker's knowledge has been moved up in tree(close to aggressive spread) they are now easier to summon and have more health.
balance: You cannot place floor tiles on rusted turfs anymore, (use a welder to scrape off the rust first).
balance: Rustbringer's oath has been reworked to propagate around in a circular fashion and not just on the bridge z level.
balance: Resist cold trait now gives immunity to freeze effects, and has now been added to Rustbringer's oath along with slowdown immunity.
fix: Fixed glowing runes being invisible.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first: cl: if you want to overwrite your GitHub username as an author ingame. -->

EnterTheJake committed:
REBOOTER IMPLANT REWORK

EnterTheJake committed:
src changed to REF(src)

EnterTheJake committed:
Merge branch 'master' into REBOOTER-REBOOTED

EnterTheJake committed:
Merge branch 'master' into REBOOTER-REBOOTED

EnterTheJake committed:
Update code/modules/surgery/organs/internal/cyberimp/augments_internal.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

EnterTheJake committed:
organ fail replaced with a simple cooldown, stun cap changed to 1 second duration upped to 6 seconds.

EnterTheJake committed:
Merge branch 'REBOOTER-REBOOTED' of https://github.com/EnterTheJake/tgstation-1 into REBOOTER-REBOOTED

EnterTheJake committed:
lintcrash

EnterTheJake committed:
sanity fix and cooldown increased to 60 seconds

EnterTheJake committed:
Merge branch 'master' into REBOOTER-REBOOTED

EnterTheJake committed:
Added sparks + code cleanup

EnterTheJake committed:
Merge branch 'master' into REBOOTER-REBOOTED

EnterTheJake committed:
stamcrit signal replaces knockdown signal

EnterTheJake committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into Spraycan-fix

EnterTheJake committed:
Spraycan Fix

EnterTheJake committed:
AAAAAAAH

EnterTheJake committed:
Rust_rework

EnterTheJake committed:
RUST_REWORK

excalisword commented:
Are bronze walls and floors also gonna be rustable?

EnterTheJake commented:
Forgot to add 

> Are bronze walls and floors also gonna be rustable?

Yes.

MrMelbert commented on code/datums/elements/rust.dm:
```diff
@@ -17,14 +17,19 @@
 	ADD_TRAIT(target, TRAIT_RUSTY, ELEMENT_TRAIT(type))
 	RegisterSignal(target, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(apply_rust_overlay))
 	RegisterSignal(target, COMSIG_ATOM_EXAMINE, PROC_REF(handle_examine))
+	RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(on_entered))
+	RegisterSignal(target, COMSIG_ATOM_EXITED, PROC_REF(on_exited))
```
This makes *all* rust apply the heretic debuff. Is that your intention?

MrMelbert commented on code/datums/status_effects/debuffs/rust_corruption.dm:
```diff
@@ -0,0 +1,16 @@
+	id = "rust_turf_effects"
+	tick_interval = 2 SECONDS
+	remove_on_fullheal = TRUE
+	status_type = STATUS_EFFECT_REFRESH
```
Using `STATUS_EFFECT_REFRESH` is redundant here, because you've have it set to the default status effect duration - infinite duration. Meaning refreshing it does nothing.

MrMelbert commented on code/datums/status_effects/debuffs/rust_corruption.dm:
```diff
@@ -0,0 +1,16 @@
+
+/datum/status_effect/rust_corruption/tick(seconds_between_ticks)
+	if(issilicon(owner))
+		var/mob/living/silicon/silicon_victim = owner
+		silicon_victim.adjustBruteLoss(10)
```
1. You don't need to cast this. `adjustBruteLoss` is in the living level. 
2. Might as well make this scale by `seconds_between_ticks`.
```suggestion
		owner.adjustBruteLoss(5 * seconds_between_ticks)
```

MrMelbert commented on code/datums/status_effects/debuffs/rust_corruption.dm:
```diff
@@ -0,0 +1,16 @@
+		var/mob/living/silicon/silicon_victim = owner
+		silicon_victim.adjustBruteLoss(10)
+		return
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/victim = owner
+	victim.adjust_disgust(3, DISGUST_LEVEL_VERYDISGUSTED)
```
Likewise, `adjust_disgust` is on the `/mob` level. So no casting necessary. 
And in fact, it does nothing if called on a non-carbon, so we don't even need to early return. 
```suggestion
	owner.adjust_disgust(3, DISGUST_LEVEL_VERYDISGUSTED)
```

MrMelbert commented on code/game/objects/items/crayons.dm:
```diff
@@ -775,7 +775,7 @@
 	. = ..()
 
-	if(!user.can_perform_action(src, NEED_DEXTERITY|NEED_HANDS))
+	if(!user.can_perform_action(src, NEED_DEXTERITY|NEED_HANDS|SILENT_ADJACENCY))
```
Snuck in from your other PR? Gotta clean your master branch.

EnterTheJake replied:
WHOOOPS.

MrMelbert commented on code/game/objects/items/crayons.dm:
```diff
@@ -786,7 +786,7 @@
 	. = ..()
 
-	if(!user.can_perform_action(src, NEED_DEXTERITY|NEED_HANDS))
+	if(!user.can_perform_action(src, NEED_DEXTERITY|NEED_HANDS|SILENT_ADJACENCY))
```
Samme

MrMelbert commented on code/game/turfs/open/asteroid.dm:
```diff
@@ -27,6 +27,7 @@
 	var/dug = FALSE
 	/// Percentage chance of receiving a bonus worm
 	var/worm_chance = 30
+	rust_resistance = RUST_RESISTANCE_ORGANIC
```
Make sure all of these are *before* new vars are declared

IE, put it here:
```dm
/turf/myturf
    name = "My turf"
    rust_resistance = RUST_RESISTANCE_ORGANIC
    var/worm_chance = 50
```
not here:
```dm
/turf/myturf
    name = "My turf"
    var/worm_chance = 50
    rust_resistance = RUST_RESISTANCE_ORGANIC
```

MrMelbert commented on code/game/turfs/open/floor/catwalk_plating.dm:
```diff
@@ -19,6 +19,7 @@
 	var/covered = TRUE
 	var/catwalk_type = "maint"
 	var/static/list/catwalk_underlays = list()
+	rust_resistance = RUST_RESISTANCE_BASIC
```
Another place for example

moocowswag commented on code/modules/reagents/chemistry/reagents/impure_reagents.dm:
```diff
@@ -104,6 +104,9 @@
 
 /datum/reagent/inverse/cryostylane/on_mob_add(mob/living/carbon/affected_mob, amount)
 	. = ..()
+	if(HAS_TRAIT(affected_mob, TRAIT_CRYOGELIDIAIMMUNE))
+		holder.remove_reagent(type, volume)
+		return
```
you could probably use cryogelidia to heal on rust tiles while safely encased in ice, im not sure if the stasis stops rust healing.

moocowswag commented:
10x rust spread to cover the station in 4 minutes instead of 40 might be too much.

What about 8x so it takes 6 minutes, that means it will likely still reach the evac shuttle unless its emergency launched and thats without you being there spreading it yourself.

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

DaydreamIQ commented:
Ascension buff is a godsend. But I wonder if there isn't a better way, wizards have that whole convert the station ritual which does something similar but its WAY more efficient. Is there not some way to borrow from that?

Xackii commented:
Why disgust instead of toxins? Sounds as useless as brain damage to the moon heretic.

EnterTheJake commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -321,7 +352,7 @@
 /datum/rust_spread
 	/// The rate of spread every tick.
-	var/spread_per_sec = 6
+	var/spread_per_sec = 60
```
it didn't seem to lag the server even with 180 spreadrate, i think it'll be PROBABLY fine.

Loafin34 commented:
I like this: what if we could reanimate rat kings into becoming our peons? akin to a miniboss-rust walker.

DaydreamIQ commented:
> Less burst in exchange for a lingering debuff that really messes you up if you overextend, fits a territorial antagonist more.
The debuff from disgust isn't really all that bad considering all you need to counter it is a glass of Sol Dry which can be obtained at any drinks vendor or a bartender. At least with toxins its difficult to counter since multiver actually has downsides to using it

Xackii commented:
> > Why disgust instead of toxins? Sounds as useless as brain damage to the moon heretic.
> 
> Less burst in exchange for a lingering debuff that really messes you up if you overextend, fits a territorial antagonist more.

if disgust is superimposed along with toxins, then this is fine. But disgust from the mark and blades? no one will take such upgrades

the-orange-cow commented:
Idea: once ascended(or earlier if you feel fit), tiles rusted by the heretic/minion/ascention rune gradually produce miasma.

EnterTheJake commented:
> > Less burst in exchange for a lingering debuff that really messes you up if you overextend, fits a territorial antagonist more.
> 
> The debuff from disgust isn't really all that bad considering all you need to counter it is a glass of Sol Dry which can be obtained at any drinks vendor or a bartender. At least with toxins its difficult to counter since multiver actually has downsides to using it

i think that's a valid point, i might have rust corruption purge

> sted by the heretic/minion/ascention rune gradually produce miasma.
> 
> Edit: this would probably necessitate giving the heretic immunity to miasmas effects if they wouldn't already have that.

yeah that's fair, i'll make it so standing on rust as a non heretic purges your chems, that should solve it.

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

EnterTheJake committed:
Melbert review + Ascension rework.

EnterTheJake committed:
Merge branch 'Rust-Rework' of https://github.com/EnterTheJake/tgstation-1 into Rust-Rework

EnterTheJake commented:
Ok, the ascension is now fixed, and I've also updated the disgust to be much higher as it was frankly underwhelming, plus I've added a chem purge effect to prevent disgust purging cheese while on rust.

newfren commented:
gonna be funny seeing people completely hardstunned on a rust tile

EnterTheJake committed:
Pulse of entropy and rust_corruption buff.

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

EnterTheJake committed:
Ascension RErework + purge fix.

EnterTheJake committed:
Merge branch 'Rust-Rework' of https://github.com/EnterTheJake/tgstation-1 into Rust-Rework

EnterTheJake committed:
_traits globalvar

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

Sakamoto4ka commented:
Wanna steal id but rust mark destroy pda moment

EnterTheJake committed:
You can't place floor tiles on rusted turf anymore.

EnterTheJake committed:
added comment and fixed a typo

EnterTheJake commented:
Last update,(hopefully) i've made it so you can't place floor tiles on rust, you'll have to scrape the rust off first.

MrMelbert commented on code/datums/elements/rust.dm:
```diff
@@ -72,3 +74,39 @@
+/datum/element/rust/proc/on_attackby(datum/source, obj/item/item, mob/user)
+	SIGNAL_HANDLER
+	if(istype(item, /obj/item/stack/tile))
+		user.balloon_alert(user, "floor too rusted")
```
```suggestion
		user.balloon_alert(user, "floor too rusted!")
```

MrMelbert commented on code/datums/elements/rust.dm:
```diff
@@ -17,6 +17,7 @@
 	ADD_TRAIT(target, TRAIT_RUSTY, ELEMENT_TRAIT(type))
 	RegisterSignal(target, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(apply_rust_overlay))
 	RegisterSignal(target, COMSIG_ATOM_EXAMINE, PROC_REF(handle_examine))
+	RegisterSignal (target, COMSIG_ATOM_ATTACKBY, PROC_REF(on_attackby))
```
Use the `item_interaction` signals rather than `attackby`.

MrMelbert commented on code/datums/status_effects/debuffs/rust_corruption.dm:
```diff
@@ -0,0 +1,12 @@
+	if(issilicon(owner))
+		owner.adjustBruteLoss(10 * seconds_between_ticks)
+		return
+	owner.adjust_disgust(10)
```
```suggestion
	owner.adjust_disgust(5 * seconds_between_ticks)
```

MrMelbert commented on code/datums/status_effects/debuffs/rust_corruption.dm:
```diff
@@ -0,0 +1,12 @@
+		owner.adjustBruteLoss(10 * seconds_between_ticks)
+		return
+	owner.adjust_disgust(10)
+	owner.reagents.remove_all(1.5)
```
```suggestion
	owner.reagents?.remove_all(0.75 * seconds_between_ticks)
```
This would presumably runtime for non-carbons stepping over rust

MrMelbert commented on code/game/atom/atom_act.dm:
```diff
@@ -201,10 +201,18 @@
 /atom/proc/rust_heretic_act()
 	return
 
+/mob/living/proc/do_rust_heretic_act(atom/target)
```
Auto-doc for what this does

MrMelbert commented on code/game/atom/atom_act.dm:
```diff
@@ -201,10 +201,18 @@
+	var/datum/antagonist/heretic/heretic_data = IS_HERETIC(src)
+	target.rust_heretic_act(heretic_data?.rust_strength)
+
+/mob/living/rust_walker/do_rust_heretic_act(atom/target)
```
Is this the right typepath? I would think it's `/mob/living/simple_animal/rust_walker` or `/mob/living/basic/rust_walker`

MrMelbert commented on code/game/turfs/turf.dm:
```diff
@@ -88,6 +88,8 @@ GLOBAL_LIST_EMPTY(station_turfs)
 	///whether or not this turf forces movables on it to have no gravity (unless they themselves have forced gravity)
 	var/force_no_gravity = FALSE
 
+	var/rust_resistance = RUST_RESISTANCE_ORGANIC
```
Auto-doc

MrMelbert commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -239,6 +262,9 @@
 		TRAIT_STUNIMMUNE,
+		TRAIT_IGNOREDAMAGESLOWDOWN,
+		TRAIT_IGNORESLOWDOWN,
+		TRAIT_FREEZEIMMUNITY,
```
ABC order

MrMelbert commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -265,10 +291,44 @@
 		color_override = "pink",
 	)
-	new /datum/rust_spread(loc)
+	//new /datum/rust_spread(loc)
```
Don't keep in commented out code

Likewise if `/datum/rust_spread` is unused now delet it

MrMelbert commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -265,10 +291,44 @@
+	var/first_third = turfs.Copy(1, round(length(turfs) * 0.33))
+	var/second_third = turfs.Copy(round(length(turfs) * 0.33), round(length(turfs) * 0.66))
+	var/third_third = turfs.Copy(round(length(turfs) * 0.6), length(turfs))
+	for(var/turf/turf as anything in first_third)
+		turf.rust_heretic_act(5)
```
```suggestion
	for(var/turf/turf as anything in first_third)
		turf.rust_heretic_act(5)
		CHECK_TICK
```
`CHECK_TICK` will essentially delay calling this to the next tick if the server's too laggy

MrMelbert commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -265,10 +291,44 @@
+	var/third_third = turfs.Copy(round(length(turfs) * 0.6), length(turfs))
+	for(var/turf/turf as anything in first_third)
+		turf.rust_heretic_act(5)
+	for(var/turf/turf as anything in second_third)
+		addtimer(CALLBACK(turf, TYPE_PROC_REF(/atom, rust_heretic_act), 5), 5 SECONDS * 0.33)
+	for(var/turf/turf as anything in third_third)
+		addtimer(CALLBACK(turf, TYPE_PROC_REF(/atom, rust_heretic_act), 5), 5 SECONDS * 0.66)
```
This is an insane amount of timer spam, Like up to 500 timers or w/e 

Rather than making 500 times to transform every single turf, just have 1 timer to transform all the turfs, IE via another proc
```dm
	addtimer(CALLBACK(src, PROC_REF(delay_transform_turfs)), 5 SECONDS * 0.33)
```
```dm
/datum/heretic_knowledge/ultimate/rust_final/proc/delay_transform_turfs(list/turfs)
	for(var/turf/turf as anything in first_third)
		turf.rust_heretic_act(5)
		CHECK_TICK
```

MrMelbert commented on code/modules/antagonists/heretic/knowledge/rust_lore.dm:
```diff
@@ -265,10 +291,44 @@
+
+/datum/heretic_knowledge/ultimate/rust_final/proc/transform_area(list/turfs)
+	turfs = shuffle(turfs)
+	var/first_third = turfs.Copy(1, round(length(turfs) * 0.33))
+	var/second_third = turfs.Copy(round(length(turfs) * 0.33), round(length(turfs) * 0.66))
+	var/third_third = turfs.Copy(round(length(turfs) * 0.6), length(turfs))
```
Just to make this a bit easier to parse
```suggestion
	var/numturfs = length(turfs)
	var/first_third = turfs.Copy(1, round(numturfs * 0.33))
	var/second_third = turfs.Copy(round(numturfs * 0.33), round(numturfs * 0.66))
	var/third_third = turfs.Copy(round(numturfs * 0.6), numturfs))
```

MrMelbert commented on code/modules/antagonists/heretic/knowledge/side_rust_cosmos.dm:
```diff
@@ -20,22 +20,21 @@
 	gain_text = "Reality begins to whisper to me. To give it its entropic end."
 	required_atoms = list(
-		/obj/item/stack/sheet/iron = 20,
-		/obj/item/trash = 2
```
Maybe keep like... *1* trash item required, so it's not SO spammable

MrMelbert commented:
Also: 
1. Mark resolved reviews as resolved please
2. Document all the changes in your CL, like the floor tile thing

EnterTheJake committed:
Update code/modules/antagonists/heretic/knowledge/rust_lore.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

EnterTheJake committed:
Melbert_Review.

EnterTheJake committed:
Merge branch 'Rust-Rework' of https://github.com/EnterTheJake/tgstation-1 into Rust-Rework

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

EnterTheJake committed:
Rust Walker Health update.

EnterTheJake committed:
Merge conflicts

MrMelbert reviewed (commented):
last review probably

EnterTheJake committed:
Review + Glowing rune fix

EnterTheJake committed:
Rune duration fix.

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

EnterTheJake committed:
Removes Temp from glowing runes obj

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

EnterTheJake committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into Rust-Rework

EnterTheJake committed:
typo

EnterTheJake committed:
Merge branch 'master' into Rust-Rework

MrMelbert reviewed (approved):
Probably fine

EnterTheJake committed:
Final-commit

MrMelbert commented on code/game/turfs/closed/wall/mineral_walls.dm:
```diff
@@ -291,16 +307,23 @@
 	smoothing_flags = SMOOTH_BITMASK | SMOOTH_DIAGONAL_CORNERS
 	canSmoothWith = SMOOTH_GROUP_SHUTTLE_PARTS + SMOOTH_GROUP_AIRLOCK + SMOOTH_GROUP_WINDOW_FULLTILE + SMOOTH_GROUP_TITANIUM_WALLS
-
+	rust_resistance = RUST_RESISTANCE_TITANIUM
```
```suggestion
	rust_resistance = RUST_RESISTANCE_TITANIUM

```

MrMelbert commented on code/game/turfs/closed/wall/mineral_walls.dm:
```diff
@@ -291,16 +307,23 @@
 	base_icon_state = "survival_pod_walls"
 	smoothing_flags = SMOOTH_BITMASK
-
+	rust_resistance = RUST_RESISTANCE_TITANIUM
```
```suggestion
	rust_resistance = RUST_RESISTANCE_TITANIUM

```

MrMelbert committed:
Assuming direct control
